### ISSUE TITLE
Tabstop documentation lacking terminfo mention

### ISSUE THREAD

**Issue by dnkl** [jwilm/alacritty#1482]
Which operating system does the issue occur on?
Linux

If on linux, are you using X11 or Wayland?
X11

On the scrollback branch, commit [23601fb](https://github.com/jwilm/alacritty/commit/23601fb), I'm seeing weird cursors jumps in Emacs 26.1. Note that this is a very (at the time of writing this) recent commit, where master was just merged).

The following two ways seem able to reproduce every time:

1. hit C-x-f (find file), at the suggested path (could be e.g. ~/, or /tmp), hit C-a. The cursor moves to the beginning of the path. So far so good. Now hit C-e. The cursor _should_ move to the end of the line, but in my case, it jumps forward a single character. After this, it appears the physical position of the cursor and the logical position is no longer synchronized, and hitting e.g. C-a again will move the cursor into the "find file" prompt (which should not be possible).

2. In an empty buffer (you could for example start Emacs with "emacs /tmp/foo.txt"), type "hello world". At the second 'o' (as in, you've typed "hello w"), the cursor jumps to the first 'o' (in "hello"). If you continue typing, the cursor will jump back. But the resulting text will be "hellr wo ld".

I've tried with various settings for the TERM variable, with no improvements. The same steps to reproduce works (i.e. does not reproduce...) in multiple other terminals (xterm, st, konsole), meaning it _shouldn't_ be a problem with my Emacs setup or similar.

I've *also* verified the issue is still there when running Emacs without an init file (emacs -q).

**Comment by chrisduerr** [comment]
Thanks for reporting this. Would you mind testing exactly which commit broke this? It's very unlikely that the commit you've linked is actually the culprit, since that was just some README change. Also if the commit in question is one of the master merge commits, it would be helpful if you could test this against the latest master, to check if this regression was due to an unclean merge or because a regression was introduced on master.

**Comment by dnkl** [comment]
TL:DR PEBKAC

The problem existed in master as well, so I did a bisect there. Turned out to be https://github.com/jwilm/alacritty/commit/650b5a0cbaccc6de2b53240372c2be79739d5d12, which didn't make any sense.

But, it pointed me in the correct direction; turned out to be that I had set tabspaces=4, which apparently was a bad idea.

FYI, st has the following to say about tabspaces:

> When you are changing this value, don't forget to adapt the »it« value in
> the st.info and appropriately install the st.info in the environment where
> you use this st version.
> 
>   it#$tabspaces,

Sure enough, if I changed tabspaces to 4 in st, without updating the terminfo, st exhibited the exact same problem,

This makes me think that perhaps the tabspaces option should be removed altogether (since we can't update terminfo at run-time), or at least the fact that it will break certain applications could be mentioned in the default configuration file.

I am fine with closing this ticket without doing anything at all, but I think it would be nice if at least the documentation of the tabspaces option could be updated.

**Comment by dnkl** [comment]
Perhaps something like:

> Note that changing the default value will break certain applications (for example, Emacs), unless a corresponding terminfo entry is used. If you do not understand what this means, do not change this.

Or is that considered too harsh?

**Comment by chrisduerr** [comment]
@dnkl I don't think the last sentence `If you do not understand what this means, do not change this.` does not add much, so I'd just leave it out. It's implicated anyways that when something might break things and you change it, you shouldn't be surprised about breakage.

However, rather than saying that changing this will break certain applications unless a corresponding terminfo entry is used, I'd just say that to make sure applications know about the altered tab width, it might be required to change the terminfo too.

I think something like this would be good:
```yml
# Spaces per Tab
#
# This setting defines the width of a tab in cells.
# Changes to this value require a restart to take effect.
#
# Since some applications (like Emacs) rely on knowing about the tab width,
# it might also be required to change the `it` value in the terminfo
# when altering this setting.
tabspaces: 8
```

Feel free to send a PR if you think this is an improvement. Also feel free to alter my version if you think you've got a better idea, this is just a rough idea of how I would word it.

**Comment by dnkl** [comment]
> It's implicated anyways that when something might break things and you change it, you shouldn't be surprised about breakage.

Sometimes people miss implicit things :)

That said, your version is much better. Being a programmer, I already hate write documentation, but not being a native english speaker (or writer) makes it so much worse.

I'll try to set up a PR "soonish" (as in, not today :) )
